### ISSUE TITLE
[Bug] Non-atomic train/valid split permanently loses the validation slice if interrupted

### ISSUE THREAD

**Issue by addyCooks** [Nano-Collective/nanotune#162]
## Description

`splitTrainValidation` (`src/lib/data.ts:767`) creates the validation set by
rewriting **two** files in sequence, with no atomicity across the pair and none
inside either write:

```ts
// src/lib/data.ts:791-792
saveTrainingData(trainExamples, false);   // truncates train.jsonl to the 90%
saveTrainingData(validExamples, true);    // then writes the other 10%
```

Between those two statements the 10% held back for validation exists **only in
memory**. It has already been removed from `train.jsonl` and has not yet landed
in `valid.jsonl`. If the process stops in that window Ctrl+C, a full disk, an
EACCES/EISDIR on the second path, a crash those examples are gone. There is
no backup and no way to reconstruct them.

## Root Cause

Two nested durability gaps.

**1. No atomicity across the pair.** The two writes are independent
`writeFileSync` calls. Nothing stages them and nothing rolls the first back if
the second fails.

**2. No atomicity within either write.** `saveTrainingData`
(`src/lib/data.ts:90-98`) is a bare truncate-then-write:

```ts
export function saveTrainingData(examples: TrainingExample[], isEval = false): void {
	ensureDataDir();
	const path = isEval ? getEvalDataPath() : getTrainDataPath();
	const content = `${examples.map(ex => JSON.stringify(ex)).join('\n')}\n`;
	writeFileSync(path, content);
}
```

So there is a second, smaller loss window inside each of the two calls.

This matters because of *when* it runs. `ensureValidationSet`
(`src/lib/data.ts:800`) calls the split only when `valid.jsonl` is empty and
`train.jsonl` is not i.e. on the **first `nanotune train` in a project** and
`train.tsx:282` invokes it immediately after the dry-run check and immediately
before the model download. That is precisely the moment a user is most likely to
hit Ctrl+C, because the next thing on screen is a multi-gigabyte download and
it's the natural point to realise a flag was wrong.

The codebase already has the helper this needs. `writeFileAtomic`
(`src/lib/config.ts:218`) does temp-write plus `renameSync`, and
`benchmark.tsx:797` already uses it for result files. The dataset writes  the
one file in the project the user cannot regenerate do not.

## Environment

- **macOS version**: N/A platform-independent (pure fs logic)
- **Apple Silicon**: N/A
- **Node version**: v22.22.3
- **Python version**: N/A
- **Nanotune version**: 1.7.0 confirmed on `main` @ `946c488`

## Steps to Reproduce

The timing window is small, so the deterministic way to demonstrate it is to
force the second write to fail rather than race it:

1. `nanotune init` a project and add 10 examples to `.nanotune/data/train.jsonl`.
2. Make the second write fail by occupying its path with a directory:
   `mkdir .nanotune/data/valid.jsonl`
3. Run `nanotune train`.
4. Inspect `.nanotune/data/train.jsonl`.

Observed `train.jsonl` now holds 9 examples. The 10th was removed from it by
the first write and the second write threw `EISDIR` before writing it anywhere.
The example is unrecoverable.

The same end state is reachable without the contrived failure by pressing Ctrl+C
between the two writes, or by hitting ENOSPC on the second.

## Expected Behavior

The split either fully applies or fully does not. An interruption at any point
leaves a complete, self-consistent pair of files worst case the pre-split
state, in which nothing has been lost.

## Actual Behavior

An interruption between `data.ts:791` and `:792` permanently destroys the
validation slice: removed from `train.jsonl`, never written to `valid.jsonl`,
held nowhere else.

## Logs/Screenshots

N/A, the loss is silent. `splitTrainValidation` returns
`{trainCount, validCount}` computed from the in-memory arrays before either
write, so on the failure path nothing on screen reflects what actually reached
disk.

## Additional Context

**Suggested fix**

1. Route `saveTrainingData` through the existing `writeFileAtomic`
   (`src/lib/config.ts:218`) instead of `writeFileSync`. This closes the
   inner window for every dataset write in the codebase, not just the split.
2. For the pair: write both to temp paths, then rename both into place. Two
   renames is not a single atomic operation, but it shrinks the window to the
   gap between two `renameSync` calls and the worst outcome becomes a
   stale-but-complete pair rather than a lost slice.
3. Minimal alternative if (2) is judged too invasive: **reverse the order** —
   write `valid.jsonl` first, truncate `train.jsonl` only after it lands. That
   alone removes the loss window entirely, leaving only a recoverable duplicate
   (examples present in both files) as the worst interruption outcome.

**Not a duplicate of #63** (closed, *Validation set is invisible and `train`
silently rewrites train.jsonl*). #63 treats the rewrite as a **visibility**
problem the user did not know their training file had been changed — and its
description states outright that "Nothing was lost." That is true on the happy
path and is exactly what this issue disputes on the interrupted path. The
durability window is not described anywhere in #63 or its fix.

**Related but distinct from #133** #133 is about `data validate --fix`
calling `saveTrainingData` more often than the user's input justifies. This
issue is about a single intended call pair not being crash-safe. They share
`saveTrainingData`'s missing atomicity, so fix (1) above would benefit both.

- [x] I have searched existing issues to ensure this is not a duplicate
- [x] I can reproduce this issue consistently
- [x] This issue occurs with the latest version of nanotune

**Comment by addyCooks** [comment]
Hey @akramcodez, 
dug into this one and the window checks out saveTrainingData is a bare truncate-then-write, 
and the split fires both calls back to back with the held-back slice living only in memory in between.

Planning to route dataset writes through the existing writeFileAtomic and stage the pair as temp-write + rename, 
with AVA coverage for the interrupted-second-write path (deterministic via the EISDIR repro, no racing).

Will keep it scoped so it doesn't collide with #133.

Could you assign this to me?

**Comment by akramcodez** [comment]
go for it @addyCooks
